src/Mod/Path/PathScripts/PathVcarve.py

```diff
@@ -355,6 +362,9 @@ def opSetDefaultValues(self, obj, job):
             else:
                 obj.OpFinalDepth = -0.1
 
+    def isToolSupported(self, obj, tool):
+        '''isToolSupported(obj, tool) ... returns True if v-carve op can work with tool.'''
+        return hasattr(tool, 'Diameter') and hasattr(tool, 'CuttingEdgeAngle') and hasattr(tool, 'TipDiameter')
```

Comment:
Do you need to check to make sure CuttingEdgeAngle is non-zero (non-90°?), or is that guaranteed? I've never looked to see what would happen if you set values for the cutting edge angle for something that wasn't a V-carving bit.

Reply:
sliptonic added an explicit check for that in the algorithm.  The `isToolSupported` is more a check if the tool can be used in theory - if this returns `False` the tool is not showing up in the drop down list and can't even be selected.